docs/sql-migration-guide.md

```diff
@@ -52,7 +52,7 @@ license: |
   
   - In Spark 3.1, the `schema_of_json` and `schema_of_csv` functions return the schema in the SQL format in which field names are quoted. In Spark 3.0, the function returns a catalog string without field quoting and in lower case. 
  
-  - In Spark 3.1, when `spark.sql.legacy.transformationPadNullWhenValueLessThenSchema` is true, Spark will pad NULL value when script transformation's output value size less then schema size in default-serde mode(script transformation with row format of `ROW FORMAT DELIMITED`). If false, Spark will keep original behavior to throw `ArrayIndexOutOfBoundsException`.
+  - In Spark 3.1, when script transformation output's value size is less then schema size in default-serde mode(script transformation with row format of `ROW FORMAT DELIMITED`), Spark will pad NUll value to supplement data. In Spark 3.0 or earlier, Spark will do nothing and throw `ArrayIndexOutOfBoundsException`. To restore the behavior before Spark 3.1, you can set `spark.sql.legacy.transformationNotPadNullToSupplementData.enabled` to `true`.
```

Comment:
I do think throwing `ArrayIndexOutOfBoundsException` is a bug. Do we still throw it somewhere in the script transformation code path?

Reply:
I think it's arguable to call it a bug. That's why I suggested to add the migration guide here. The behaviour is just mismatch compared to Hive's behaviour. The previous behaviour was not padding null with the failfast.

Reply:
If we intend to fail, we should give a better error message with `AnalysisException`, not bug-like exceptions such as `ArrayIndexOutOfBoundsException`, NPE, etc.

Reply:
Yeah that might be the best for completeness. I am just elaborating why I think it should be best to update the migration guide. We'll move to this new behaviour by default and probably remove it away in the future anyway.

Reply:
Yea I'm just saying that we usually don't add migration guide if we make something failed to work, as it's not a breaking change and users have nothing to do for migration.

Reply:
So, What should I do next? Personally, we need to let users know about this change

Reply:
Nit: NUll -> NULL
Yeah, why would you ever set this flag to the original behavior, to fail? this is just a bug fix. Nobody can reasonably be relying on this behavior.

Reply:
All  right, we  can just revert last pr.